replace MSVC-only Sleep and GetTickCount

q4a opened the pull request.

q4a committed:
replace MSVC-only Sleep, GetTickCount

espkk reviewed (changes requested):
Thanks for the PR! 
Unfortunately, neither `system_clock` nor `time()` are suitable replacements for ticks counter query. I can replace it with a cross-platform wrapper a bit later.
Replacing `Sleep` with `sleep_for` is not a problem though

q4a commented:
> I can replace it with a cross-platform wrapper a bit later.

Thanks, I will wait impatiently

> Replacing `Sleep` with `sleep_for` is not a problem though

I will update PR and leave only `Sleep` replacing.

q4a commented:
I did some tests with `std::srand`. Looks like it's better replace
`srand(GetTickCount());`
with
`srand(duration_cast<milliseconds>(system_clock::now().time_since_epoch()).count());`

Here is test code:
```
#ifdef _WIN32
#include <Windows.h>
#endif
#include <ctime>
#include <iostream>
#include <thread>

using std::chrono::duration_cast;
using std::chrono::milliseconds;
using std::chrono::system_clock;

int main()
{
    int random_variable;
    for (auto i = 1; i <= 2; i++)
    {
#ifdef _WIN32
        for (auto j = 1; j <= 2; j++)
        {
            std::srand(GetTickCount());
            random_variable = std::rand();
            std::cout << "Random value with GetTickCount on [0 " << RAND_MAX << "]: " << random_variable << '\n';
        }
#endif
        for (auto j = 1; j <= 2; j++)
        {
            srand(duration_cast<milliseconds>(system_clock::now().time_since_epoch()).count());
            random_variable = std::rand();
            std::cout << "Random value with system_clock on [0 " << RAND_MAX << "]: " << random_variable << '\n';
        }
        for (auto j = 1; j <= 2; j++)
        {
            srand(std::time(nullptr));
            random_variable = std::rand();
            std::cout << "Random value with std::time on [0 " << RAND_MAX << "]: " << random_variable << '\n';
        }
        std::this_thread::sleep_for(milliseconds(5));
    }
}
```
Results on Windows:
```
Random value with GetTickCount on [0 32767]: 14268
Random value with GetTickCount on [0 32767]: 14268
Random value with system_clock on [0 32767]: 15125
Random value with system_clock on [0 32767]: 15134
Random value with std::time on [0 32767]: 7785
Random value with std::time on [0 32767]: 7785
Random value with GetTickCount on [0 32767]: 14317
Random value with GetTickCount on [0 32767]: 14317
Random value with system_clock on [0 32767]: 15174
Random value with system_clock on [0 32767]: 15174
Random value with std::time on [0 32767]: 7785
Random value with std::time on [0 32767]: 7785
```
Results on Linux:
```
$ clang++-13 -std=c++20 Source.cpp
$ ./a.out
Random value with system_clock on [0 2147483647]: 1824576723
Random value with system_clock on [0 2147483647]: 1824576723
Random value with std::time on [0 2147483647]: 1351709256
Random value with std::time on [0 2147483647]: 1351709256
Random value with system_clock on [0 2147483647]: 306497496
Random value with system_clock on [0 2147483647]: 306497496
Random value with std::time on [0 2147483647]: 1351709256
Random value with std::time on [0 2147483647]: 1351709256
```

q4a committed:
Return GetTickCount() for ticks counter query +
replace srand(std::time(nullptr)) with
srand(duration_cast<milliseconds>(system_clock::now().time_since_epoch()).count())

espkk commented on src/libs/sea_ai/src/locator_show.cpp:
```diff
@@ -155,7 +157,7 @@ void SeaLocatorShow::Realize(uint32_t Delta_Time)
     {
         bShow ^= 1;
-        Sleep(200);
+        std::this_thread::sleep_for(std::chrono::milliseconds(100));
```
```suggestion
        std::this_thread::sleep_for(std::chrono::milliseconds(200));
```
(constant from the original code)

espkk committed:
Correct sleep_for time

espkk reviewed (approved):
Everything is good, thanks!
